### PR TITLE
[CPDNPQ-909] Implement handler for UserUpdated GetAnIdentity::WebhookMessage records

### PR DESCRIPTION
**Pull request by Lockyy**
### Context

Ticket: https://dfedigital.atlassian.net/browse/CPDNPQ-909

UserUpdated webhook messages inform us of changes made to users. We need to store those changes and pass them through to ECF.

### Changes proposed in this pull request

- Add Services::GetAnIdentity::Webhooks::UserUpdatedProcessor for handling UserUpdated messags
- Begin recording trn_lookup_status from GAI login requests

### CODE REVIEW

File: app/lib/forms/qualified_teacher_check.rb [hunk 1]
```diff
@@ -117,6 +122,7 @@ def active_alert?
 
     def after_save
       wizard.store["trn_verified"] = trn_verified?
+      wizard.store["trn_lookup_status"] = trn_lookup_status
```
Comment (Lockyy) [inline app/lib/forms/qualified_teacher_check.rb:125]: Ensures consistency between non-GAI and GAI lookup status.

File: Gemfile [hunk 2]
```diff
@@ -61,6 +61,7 @@ end
 
 group :test do
   gem "factory_bot_rails"
+  gem "faker"
```
Comment (shjohnson) [inline Gemfile:64]: 🙌

**Review by shjohnson — APPROVED** [review]
Looks good!

File: app/lib/forms/qualified_teacher_check.rb [hunk 3]
```diff
@@ -107,6 +107,11 @@ def trn_verified?
       @trn_verified
     end
 
+    def trn_lookup_status
+      # This mimics the lookup status that the get an identity service would give
+      trn_verified? ? "Found" : "Failed"
```
[the remark this answers is not in the document]
Comment (Lockyy) [reply app/lib/forms/qualified_teacher_check.rb:112]: It means "We looked for their TRN in the DQT database (and other sources) and could not find it.

Get an Identiy won't provide a TRN in those situations but our side lets user's proceed with the one they entered. But the meaning is the same, we tried to look it up to verify it and we failed to find it.

File: spec/jobs/get_an_identity/process_webhook_message_job_spec.rb [hunk 4]
```diff
@@ -4,20 +4,35 @@
...
+
+    context "when the webhook message_type is UserUpdated" do
+      let(:message_type) { "UserUpdated" }
+      let(:message) { {} }
+
+      it "sends webhook message to the UserUpdatedProcessor" do
+        expect(Services::GetAnIdentity::Webhooks::UserUpdatedProcessor).to receive(:call).with(webhook_message:)
+        described_class.perform_now(webhook_message:)
+      end
+    end
```
[the remark this answers is not in the document]
Comment (Lockyy) [reply spec/jobs/get_an_identity/process_webhook_message_job_spec.rb:36]: What do you mean? Like if the message is nonsense?

[the remark this answers is not in the document]
Comment (shjohnson) [reply spec/jobs/get_an_identity/process_webhook_message_job_spec.rb:36]: Yeah i guess so, the other test mostly catches it and to be honest i think this doesnt matter

File: app/lib/services/get_an_identity/webhooks/user_updated_processor.rb [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        def initialize(webhook_message:)
+          @webhook_message = webhook_message
+        end
+
+        def call
+          return wrong_processor_failure unless webhook_message.message_type == "UserUpdated"
+          return no_user_found_failure if user.blank?
+
+          if user.updated_from_tra_at.present? && user.updated_from_tra_at > webhook_message.sent_at
+            return more_recent_data_recorded_success
```
[the remark this answers is not in the document]
Comment (shjohnson) [reply app/lib/services/get_an_identity/webhooks/user_updated_processor.rb:22]: Yeah feels like we can remove the return value and the return command

**Review by shjohnson — APPROVED** [review]

**Pull request merged**